packages/amazonq/src/api.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GenerateAssistantResponseCommandOutput, GenerateAssistantResponseRequest } from '@amzn/codewhisperer-streaming'
+import { AuthUtil } from 'aws-core-vscode/codewhisperer'
+import { ChatSession } from 'aws-core-vscode/codewhispererChat'
+import { qAPI } from 'aws-core-vscode/amazonq'
+
+export default {
+    chatApi: {
+        async chat(request: GenerateAssistantResponseRequest): Promise<GenerateAssistantResponseCommandOutput> {
+            const chatSession = new ChatSession()
+            return chatSession.chat(request)
+        },
+    },
+    authApi: {
+        async reauthIfNeeded() {
+            if (AuthUtil.instance.isConnectionExpired()) {
+                await AuthUtil.instance.showReauthenticatePrompt()
+            }
+        },
+        async getChatAuthState() {
+            return AuthUtil.instance.getChatAuthState()
+        },
+    },
+} as qAPI
```

Comment:
```suggestion
} as api
```

Also, `amazonq` is the name for "Amazon Q" in most of the codebase, let's stay consistent with that.